civis/io/_files.py

```diff
@@ -315,11 +320,22 @@ def _civis_to_file(file_id, buf, api_key=None, client=None):
         raise EmptyResultError('Unable to locate file {}. If it previously '
                                'existed, it may have '
                                'expired.'.format(file_id))
-    response = requests.get(url, stream=True)
-    response.raise_for_status()
-    chunked = response.iter_content(CHUNK_SIZE)
-    for lines in chunked:
-        buf.write(lines)
+
+    # Store the current buffer position in case we need to retry below.
+    buf_orig_position = buf.tell()
+
+    @retry(RETRY_EXCEPTIONS)
+    def _download_url_to_buf():
```

Comment:
Another way to do this without re-downloading the bytes you already retrieved is to use range headers in the HTTP Get request like [so](https://github.com/civisanalytics/bowerbird/blob/master/bowerbird/filesystem/_http.py#L88). You can probably do something similar on Put but I haven't done that before.

Up to you...

Reply:
That's a good idea, but I'm going to punt on it for now.  I filed #274.